src/Type/IntegerRangeType.php

```diff
@@ -315,6 +315,16 @@ public function isSmallerThan(Type $otherType, PhpVersion $phpVersion): TrinaryL
 			$maxIsSmaller = (new ConstantIntegerType($this->max))->isSmallerThan($otherType, $phpVersion);
 		}
 
+		// 0 can have different results in contrast to the interval edges, see https://3v4l.org/iGoti
```

Comment:
This is expected, at least for me, better to document the `-1 > null` case.

I belive `-1 > null` is somethings that can be even fixed in php-src, as I would expect `null` to be compared as `0`.

Reply:
feel free to open a php-src bug reported. I guess they won't fix it because of BC concerns

Reply:
That might be a good reason to report `>, >=, <, <=` operator with nullable integer in phpstan-strict-rule no ?
https://phpstan.org/r/ed5e293c-721b-4b14-b9ea-aa42876f8a2c